storage/cloud-client/encryption.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to upload and download encrypted blobs
+(objects) in Google Cloud Storage.
+
+Use `generate-encryption-key` to generate an example key:
+
+    python encryption.py generate-encryption-key
+
+Then use the key to upload and download files encrypted with a custom key.
+
+For more information, see the README.md under /storage and the documentation
+at https://cloud.google.com/storage/docs/encryption.
+"""
+
+import argparse
+import base64
+import os
+
+from gcloud import storage
+
+
+def generate_encryption_key():
+    """Generates a 256 bit (32 byte) AES encryption key and prints the
+    base64 representation.
+
+    This is included for demonstration purposes. You should generate your own
+    key. Please remember that encryption keys should be handled with a
+    comprehensive security policy.
+    """
+    key = os.urandom(32)
+    encoded_key = base64.b64encode(key).decode('utf-8')
+    print('Base 64 encoded encryption key: {}'.format(encoded_key))
+
+
+def upload_encrypted_blob(bucket_name, source_file_name,
+                          destination_blob_name, base64_encryption_key):
+    """Uploads a file to a Google Cloud Storage bucket using a custom
```

Comment:
I tend to prefer self-documenting parameters, without explicit arg definitions in doc strings because 1) the doc strings become a hassle to maintain, and 2) they're often unnecessary. For example, I try to avoid things like:

```
def upload_blob(bucket_name):
    """Keyword arguments:
        bucket_name: the name of the bucket to upload to
    """
```

When you have more complex input params (objects, dictionaries with variable keys), etc., then I think it becomes more worth it.

tl;dr: In this case, I don't think it's necessary, but in some more complex cases I do.